lib/openstack_query/query_blocks/query_chainer.py

```diff
@@ -12,25 +16,75 @@ class QueryChainer:
     def __init__(self, chain_mappings):
         self._chain_mappings = chain_mappings
 
-    def get_supported_queries(self) -> List[QueryTypes]:
+    def get_chaining_props(self) -> List[QueryTypes]:
         """
-        Gets a list of all supported query types that this query can be chained into
+        Gets a list of all supported props that can be used to chain to other queries
         """
         return list(self._chain_mappings.keys())
 
-    def check_supported(self, query: QueryTypes) -> bool:
+    def check_prop_supported(self, prop: PropEnum) -> bool:
         """
-        Checks whether this query can be chained into a given query
-        :param query: QueryTypes enum representing the query to check if chaining is supported for
+        Checks whether a prop can be used as a link prop for the query
+        :param prop: prop to check if it is a chain prop for given query
         """
-        return query in self._chain_mappings.keys()
+        return prop in self._chain_mappings.keys()
 
-    def get_link_props(self, query: QueryTypes) -> Optional[Tuple[PropEnum, PropEnum]]:
+    def get_link_props(
+        self, next_query: QueryTypes
+    ) -> Optional[Tuple[PropEnum, PropEnum]]:
         """
         Get the link property pair to use to convert from current query to new query
         (if mapping exists)
-        :param query: QueryTypes enum representing the query to get link properties for
+        :param next_query: next Query QueryTypes
         """
-        if not self.check_supported(query):
-            return None
-        return self._chain_mappings[query]
+
+        new_query_props = list(next_query.value.get_prop_mapping())
+        for start_chain_prop, check_prop in self._chain_mappings.items():
+            for target_prop in new_query_props:
+                if target_prop == check_prop:
+                    return start_chain_prop, target_prop
```

Comment:
Why do we validate that `target_prop` == `check_prop`
Surely if the key is in the chaining dict, we can just assume that the target is valid? Rather than iterate both dictionaries, since this effectively means we could drop the mapping all-together?